content/en/docs/tasks/configure-pod-container/migrate-from-psp.md

```diff
@@ -19,30 +19,561 @@ admission controller. This can be done effectively using a combination of dry-ru
 
 - Ensure the `PodSecurity` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features) is enabled.
 
+This page assumes you are already familiar with the basic [Pod Security Admission](/docs/concepts/security/pod-security-admission/)
+concepts.
+
 <!-- body -->
 
-## Steps
-
-- **Eliminate mutating PodSecurityPolicies, if your cluster has any set up.**
-  - Clone all mutating PSPs into a non-mutating version.
-  - Update all ClusterRoles authorizing use of those mutating PSPs to also authorize use of the
-    non-mutating variant.
-  - Watch for Pods using the mutating PSPs and work with code owners to migrate to valid,
-    non-mutating resources.
-  - Delete mutating PSPs.
-- **Select a compatible policy level for each namespace.** Analyze existing resources in the
-  namespace to drive this decision.
-  - Review the requirements of the different [Pod Security Standards](/docs/concepts/security/pod-security-standards).
-  - Evaluate the difference in privileges that would come from disabling the PSP controller.
-  - In the event that a PodSecurityPolicy falls between two levels, consider:
-    - Selecting a _less_ permissive PodSecurity level prioritizes security, and may require adjusting
-      workloads to fit within the stricter policy.
-    - Selecting a _more_ permissive PodSecurity level prioritizes avoiding disrupting or
-      changing workloads, but may allow workload authors in the namespace greater permissions
-      than desired.
-- **Apply the selected profiles in `warn` and `audit` mode.** This will give you an idea of how
-  your Pods will respond to the new policies, without breaking existing workloads. Iterate on your
-  [Pods' configuration](/docs/concepts/security/pod-security-admission#configuring-pods) until
-  they are in compliance with the selected profiles.
-- Apply the profiles in `enforce` mode.
-- Stop including `PodSecurityPolicy` in the `--enable-admission-plugins` flag.
\ No newline at end of file
+## Overall approach
+
+There are multiple strategies you can take for migrating from PodSecurityPolicy to Pod Security
+Admission. The following steps are one possible migration path, with a goal of minimizing both the
+risks of a production outage and of a security gap.
+
+<!-- Keep section header numbering in sync with this list. -->
+0. Decide whether Pod Security Admission is the right fit for your use case.
+1. Review namespace permissions
+2. Simplify & standardize PodSecurityPolicies
+3. Update namespaces
+   1. Identify an appropriate Pod Security level
+   2. Verify the Pod Security level
+   3. Enforce the Pod Security level
+   4. Bypass PodSecurityPolicy
+4. Review namespace creation processes
+5. Disable PodSecurityPolicy
+
+## 0. Decide whether Pod Security Admission is right for you
```

Comment:
For all of these numbered headers, I would change them to be like:
```markdown
## 0. Decide whether Pod Security Admission is right for you {#decide-whether-pod-security-admission-is-right-for-you}
```
which _omits_ the numbering. That then gives it a fragment identifier which is stable if we renumber.

(It also leaves room for a future shortcode or whatever that does auto-numbered headings.)

Reply:
Oh yeah, good call.